samcli/commands/init/init_templates.py

```diff
@@ -118,23 +120,72 @@ def _shared_dir_check(self, shared_dir):
             return False
 
     def _clone_repo(self):
+        if not self._auto_clone:
+            return  # Unit test escape hatch
+        # check if we have templates stored already
         shared_dir = global_cfg.config_dir
         if not self._shared_dir_check(shared_dir):
+            # Nothing we can do if we can't access the shared config directory, use bundled.
             return
         expected_path = os.path.normpath(os.path.join(shared_dir, self._repo_name))
-        if self._should_clone_repo(expected_path):
+        if self._template_directory_exists(expected_path):
+            self._overwrite_existing_templates(shared_dir, expected_path)
+        else:
+            # simply create the app templates repo
+            self._clone_new_app_templates(shared_dir, expected_path)
+        self.clone_attempted = True
+
+    def _overwrite_existing_templates(self, shared_dir, expected_path):
+        self.repo_path = expected_path
+        # workflow to clone a copy to a new directory and overwrite
+        with osutils.mkdir_temp(ignore_errors=True) as tempdir:
             try:
+                expected_temp_path = os.path.normpath(os.path.join(tempdir, self._repo_name))
+                LOG.info("\nCloning app templates from %s", self._repo_url)
                 subprocess.check_output(
-                    [self._git_executable(), "clone", self._repo_url], cwd=shared_dir, stderr=subprocess.STDOUT
+                    [self._git_executable(), "clone", self._repo_url, self._repo_name],
+                    cwd=tempdir,
+                    stderr=subprocess.STDOUT,
                 )
+                # Now we need to delete the old repo and move this one.
+                try:
```

Comment:
Suggestion: can we move this try except block which is inside another try except block to another method, helps with readability.

Reply:
Done in new version.